third_party/move/move-compiler-v2/src/pipeline/abort_analysis.rs

```diff
@@ -0,0 +1,179 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! Computes at a given program point, how the program may exit later.
+//! Does the program returns, aborts, or doesn't terminate?
+//! Check documentation of `ExitState` for more on the abstract domain used in the analysis.
+//! The analysis is intraprocedural, and considers any user function may abort.
+//!
+//! Requires: The program cannot silently exits, i.e., neither returns nor aborts, but runs out of instructions.
+
+use abstract_domain_derive::AbstractDomain;
+use itertools::Itertools;
+use move_binary_format::file_format::CodeOffset;
+use move_model::model::FunctionEnv;
+use move_stackless_bytecode::{
+    dataflow_analysis::{DataflowAnalysis, TransferFunctions},
+    dataflow_domains::{AbstractDomain, JoinResult, SetDomain},
+    function_target::{FunctionData, FunctionTarget},
+    function_target_pipeline::{FunctionTargetProcessor, FunctionTargetsHolder},
+    stackless_bytecode::Bytecode,
+    stackless_control_flow_graph::StacklessControlFlowGraph,
+};
+use std::{collections::BTreeMap, fmt::Display};
+
+/// The power set lattice of `ExitStatus`
+///
+/// - the join operation is set union
+/// - the top element is { Return, Abort }: may return, abort, or not terminate
+/// - { Return }: may return or not terminate
+/// - { Abort }: may abort or not terminate
+/// - the bot element is {}: don't terminate
+///
+/// That is, if at a program point the abstract state is `s`, then from that point,
+/// the program can only exit in the exit states contained in `s`, if the program does terminate.
+#[derive(AbstractDomain, Clone)]
+pub struct ExitState(SetDomain<ExitStatus>);
+
+impl ExitState {
```

Comment:
Yep. I am waiting until the PR is approved to change the name, because we were referring this as the abort analysis.

Reply:
So, to confirm the action item here, you will rename things after the PR is approved but before it is merged in? It is generally preferred not to do so, so that unreviewed code that is merged in is minimized.

Reply:
done